pkg/k8s/pipelineRun.go

```diff
@@ -566,3 +569,27 @@ func (r *pipelineRun) mustNotHavePendingChanges() {
 		panic(fmt.Errorf("there are pending changes"))
 	}
 }
+
+// NewPipelineRunLoggingContext returns new context with
+// pipeline run's information for logging.
+func NewPipelineRunLoggingContext(
```

Comment:
Agree with moving this function to runctl/controller.go and passing logger as pass by value.

However, passing logger suits best instead of Context, because it is easier during unit-testing to pass ktesting logger instance to the function.

See -
https://github.com/SAP/stewardci-core/pull/387/files#diff-f86bb792cd003af31ff3495e8f779e767f3ce5ae9ef6a30e49233c9676377c88R60-R65

or 
```go
t.Run(test.name, func(t *testing.T) {
  ...
  logger := ktesting.NewLogger(
	t,
	ktesting.NewConfig(
	ktesting.BufferLogs(true),
	),
  )
...
}
```

This ktesting logger has an [API](https://pkg.go.dev/k8s.io/klog/v2@v2.100.1/ktesting#Underlier) implemented to access underlying logger (sink). This sink makes it easy to inspect [log entries](https://pkg.go.dev/k8s.io/klog/v2@v2.100.1/ktesting#Log) in unit-tests.

Additionally, IMO passing logger instance is not much different from passing Context.